src/layer.rs

```diff
@@ -1220,14 +1297,15 @@ where
     /// [`Span`]: opentelemetry::trace::Span
     fn on_close(&self, id: span::Id, ctx: Context<'_, S>) {
         let span = ctx.span(&id).expect("Span not found, this is a bug");
+        // Now get mutable extensions for removal
         let (otel_data, timings) = {
-            let mut extensions = span.extensions_mut();
+            let mut span_extensions = span.extensions_mut();
```

Comment:
I think this name change is confusing. This is not the `OpenTelemetrySpanExt`, but the subscriber extension data used in the `OpenTelemetryLayer`.